idea/idea-fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLConvertToBlockBodyIntention.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.fir.intentions
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.idea.fir.api.AbstractHLIntention
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicabilityRange
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicatorInputProvider
+import org.jetbrains.kotlin.idea.fir.api.applicator.inputProvider
+import org.jetbrains.kotlin.idea.fir.applicators.ApplicabilityRanges
+import org.jetbrains.kotlin.idea.formatter.adjustLineIndent
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.types.KtClassErrorType
+import org.jetbrains.kotlin.idea.frontend.api.types.KtType
+import org.jetbrains.kotlin.idea.util.resultingWhens
+import org.jetbrains.kotlin.idea.util.setType
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+
+class HLConvertToBlockBodyIntention :
+    AbstractHLIntention<KtDeclarationWithBody, HLConvertToBlockBodyIntention.Input>(KtDeclarationWithBody::class, applicator) {
+
+    class Input(
+        val returnType: KtType,
+        val returnTypeIsUnit: Boolean,
+        val returnTypeIsNothing: Boolean,
+        val returnTypeString: String,
+        val returnTypeClassId: ClassId?,
+        val bodyTypeIsUnit: Boolean,
+        val bodyTypeIsNothing: Boolean,
+        val reformat: Boolean,
+    ) : HLApplicatorInput
+
+    override fun allowCaretInsideElement(element: PsiElement) = element !is KtDeclaration && super.allowCaretInsideElement(element)
+
+    override val applicabilityRange: HLApplicabilityRange<KtDeclarationWithBody> get() = ApplicabilityRanges.SELF
+
+    override val inputProvider: HLApplicatorInputProvider<KtDeclarationWithBody, Input>
+        get() = inputProvider { psi ->
+            if (psi is KtNamedFunction) {
+                val returnType = psi.getReturnKtType()
+                if (!psi.hasDeclaredReturnType() && returnType is KtClassErrorType) return@inputProvider null
+            }
+            createInputForDeclaration(psi, true)
+        }
+
+    companion object {
+        fun KtAnalysisSession.createInputForDeclaration(declaration: KtDeclarationWithBody, reformat: Boolean): Input? {
```

Comment:
`reformat` is always `true`, do you need it?

Reply:
This is copied from https://github.com/JetBrains/kotlin/blob/3e8b0ae05abc780458f2894c3f2d4c630dec60ff/idea/src/org/jetbrains/kotlin/idea/intentions/ConvertToBlockBodyIntention.kt#L54 and this flag is used in other actions. So I am doing the same here in case we want to reuse this input in future when rewriting the other actions